### ISSUE TITLE
Latest Magisk canary fails to install on Fire HD 8 2018 

### ISSUE THREAD

**Issue by 488315** [topjohnwu/Magisk#2720]
The issue that the Magisk Installer detects the Boot Image on /dev/log/kernel and it should be saying /dev/block/mmcblk0p11 as a target boot image.
If you need me to attach more logs or files please let me know. The TWRP Version is 3.3.1-0
[recovery.log](https://github.com/topjohnwu/Magisk/files/4506902/recovery.log)
the recovery fstab is 
[recovery.fstab.txt](https://github.com/topjohnwu/Magisk/files/4506916/recovery.fstab.txt)

**Comment by topjohnwu** [comment]
@osm0sis I think this is a regression of allowing our `find` command to also search for character files

**Comment by osm0sis** [comment]
Yep, you're right. I'll add a `grep -v` to weed out any false positives we come across. :+1:

**Comment by 488315** [comment]
ok, so I have tried to make changes to the util_function script. It seems to change to use the target boot image as /dev/BOOT and it still won't install.

**Comment by osm0sis** [comment]
/dev/BOOT actually exists and is the boot partition on some MTD/NAND devices, so I'm not sure how to accomodate your false positive there.. I'll experiment I guess.

Can you give me the output of "ls -alR /dev" for your device. It's pretty atypical it seems.

**Comment by 488315** [comment]
Here you go

https://pastebin.com/j0UzXQa1

**Comment by osm0sis** [comment]
Hey, I need `ls -alR /dev` not `ls -alr /dev`; they're different.

**Comment by 488315** [comment]
[ls.txt](https://github.com/topjohnwu/Magisk/files/4518954/ls.txt)

**Comment by osm0sis** [comment]
And once more from in recovery, since there's no /dev/log/kernel in that one, and I want to make sure there aren't any more surprises in there. :wink:

**Comment by 488315** [comment]
https://pastebin.com/m7MYgAUe
Here’s the one from recovery.

**Comment by osm0sis** [comment]
@488315, alright I think I got it this time. Can you try out the changes?

@Msprg and @martobg10 can you guys try these out too? I'm pretty sure I've still got you covered (possibly covered better, since I notice @Msprg had /dev/BOOT as well), but I want to make sure. :+1:

**Comment by 488315** [comment]
Sure I can try

**Comment by 488315** [comment]
![Screenshot_2020-04-22-22-23-36](https://user-images.githubusercontent.com/19639817/80055795-75f44800-84d7-11ea-9acd-cbb58dbf4d3c.png)
[recovery.log](https://github.com/topjohnwu/Magisk/files/4520071/recovery.log)
It works to me! Thanks, @osm0sis.

**Comment by 488315** [comment]
Do you think it’s ok to close this issue? @osm0sis? 
    	
    	

    	Get Outlook for iOS
    
  




On Wed, Apr 22, 2020 at 8:27 PM -0700, "kjones" <notifications@github.com> wrote:

















recovery.log

It works to me! Thanks, @osm0sis.

**Comment by osm0sis** [comment]
No, I'm waiting on confirmation from two testers with other outlier MTD/NAND devices.

The issue will close itself once my commit is merged. So best to leave it open until then.

**Comment by Msprg** [comment]
Helllo @osm0sis , late to the party, as usual 😄 

It is still working with your latest commit 👍 , but I would like to see it working from Magisk manager too one day...

I see you literary wrote algorithm this time, nice.

Thank you for your tireless work-

**Comment by osm0sis** [comment]
@Msprg if it's still not working from Manager after these fixes get merged please open another issue and we'll dig into it. :raised_hands:

**Comment by martobg10** [comment]
Hey, @osm0sis! 😃
I can also confirm that it works with your latest commit 🎉

> ...but I would like to see it working from Magisk manager too one day...

**+1**
I also want to help with that. The latest Canary Magisk Manager thinks that I have **/dev/BOOT**, but I don't have it.

![Screenshot_2020-04-23-22-12-19](https://user-images.githubusercontent.com/44940748/80139816-bd4e0780-85af-11ea-9c95-f36ba41c07ab.png)

@Msprg, if you can, please open a new issue, because I'm not very good with English 😄

**Comment by Msprg** [comment]
Hello @martobg10 !

As @osm0sis said, we will wait until the changes get merged, then, when a new canary/beta Magisk version will be released, we try to install it using Magsik Manager directly. If it still fails we ("I") wil open a new issue and with the help of @osm0sis, with his seemingly infinite passion, (😅)  get this sorted 👍 .

P.S. Your english is very good and I am sure that a year later it will be even better, and you will not even notice the transition.

**Comment by osm0sis** [comment]
20407 should be good to go, all! :+1:

**Comment by martobg10** [comment]
@osm0sis,
Updating from 20406 to 20407 in Canary Magisk Manager shows exactly the same from the photo from my previous post.

**Comment by osm0sis** [comment]
> @osm0sis,
> Updating from 20406 to 20407 in Canary Magisk Manager shows exactly the same from the photo from my previous post.

You would need to be on 20407 first to see any difference in Manager. Try dirty installing 20407 again over 20407 and let me know.

**Comment by martobg10** [comment]
@osm0sis,
I updated from 20406 to 20407 using recovery. After that, I tried to dirty flash 20407 over 20407 using Canary Magisk Manager, but the same problem from the photo.

**Comment by osm0sis** [comment]
Okay, thanks. We'll carry on debugging that in a different issue when @Msprg gets the chance.

**Comment by Msprg** [comment]
Gonna try this NOW!
...
Flashing 20407 over 20406 Via MM ... Failed
Flashing 20407 over recovery ... Succeeded
Trying to flash 20407 over 20407 via MM ... Failed

Gonna open new issue in a short while.
